gnupg.c

```diff
@@ -1471,6 +1473,67 @@ PHP_FUNCTION(gnupg_decrypt)
 }
 /* }}} */
 
+/* {{{ proto string gnupg_messagekeys(string enctext)
+ * returns the recipient keyids and their status for the given enctext
+ */
+PHP_FUNCTION(gnupg_messagekeys)
+{
+	char *enctxt;
+	phpc_str_size_t	enctxt_len;
+	gpgme_data_t in, out;
+	gpgme_decrypt_result_t result;
+	gpgme_recipient_t recipient;
+
+	GNUPG_GETOBJ();
+
+	if (this) {
+		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s",
+				&enctxt, &enctxt_len) == FAILURE) {
+			return;
+		}
+	} else {
+		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs",
+				&res, &enctxt, &enctxt_len) == FAILURE) {
+			return;
+		}
+		GNUPG_RES_FETCH();
+	}
+
+	if (!PHP_GNUPG_DO(gpgme_data_new_from_mem(&in, enctxt, enctxt_len, 0))) {
+		GNUPG_ERR("could not create in-data buffer");
+	}
+
+	if ((PHPC_THIS->err = gpgme_data_new(&out)) != GPG_ERR_NO_ERROR) {
+		GNUPG_ERR("could not create out-data buffer");
+		gpgme_data_release(in);
+		return;
+	}
+
+	PHP_GNUPG_DO(gpgme_op_decrypt(PHPC_THIS->ctx, in, out));
+
+	result = gpgme_op_decrypt_result(PHPC_THIS->ctx);
+
+	gpgme_data_release(in);
+	gpgme_data_release(out);
+
+	if (result->recipients) {
```

Comment:
NIT: I would swap the condition check and first check if no recipients and then return false. So there is less nesting...

Reply:
Will make the change.